server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,137 +274,426 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
+        }
+
+        executorService = MoreExecutors.listeningDecorator(
+            Executors.newScheduledThreadPool(
+                lookupCoordinatorManagerConfig.getThreadPoolSize(),
+                Execs.makeThreadFactory("LookupCoordinatorManager--%s")
+            )
+        );
+
+        //Note: this call is idempotent, so multiple start() would not cause any problems.
+        lookupMapConfigRef = configManager.watch(
+            LOOKUP_CONFIG_KEY,
+            new TypeReference<Map<String, Map<String, LookupExtractorFactoryMapContainer>>>()
             {
-              final Map<String, Map<String, Map<String, Object>>> allLookupTiers = lookupMapConfigRef.get();
-              // Sanity check for if we are shutting down
-              if (Thread.currentThread().isInterrupted()) {
-                LOG.info("Not updating lookups because process was interrupted");
-                return;
+            },
+            null
+        );
+
+        // backward compatibility with 0.9.x
+        if (lookupMapConfigRef.get() == null) {
+          Map<String, Map<String, Map<String, Object>>> oldLookups = configManager.watch(
+              OLD_LOOKUP_CONFIG_KEY,
+              new TypeReference<Map<String, Map<String, Map<String, Object>>>>()
+              {
+              },
+              null
+          ).get();
+
+          if (oldLookups != null) {
+            Map<String, Map<String, LookupExtractorFactoryMapContainer>> converted = new HashMap<>();
+            for (String tier : oldLookups.keySet()) {
+              Map<String, Map<String, Object>> oldTierLookups = oldLookups.get(tier);
+              if (oldLookups != null && !oldLookups.isEmpty()) {
+                Map<String, LookupExtractorFactoryMapContainer> convertedTierLookups = new HashMap<>();
+                for (Map.Entry<String, Map<String, Object>> e : oldTierLookups.entrySet()) {
+                  convertedTierLookups.put(e.getKey(), new LookupExtractorFactoryMapContainer(null, e.getValue()));
+                }
+                converted.put(tier, convertedTierLookups);
               }
-              if (!started) {
-                LOG.info("Not started. Returning");
-                return;
+            }
+            configManager.set(
+                LOOKUP_CONFIG_KEY,
+                converted,
+                new AuditInfo("autoConversion", "autoConversion", "127.0.0.1")
+            );
+          }
+        }
+
+        this.backgroundManagerExitedLatch = new CountDownLatch(1);
+        this.backgroundManagerFuture = executorService.scheduleWithFixedDelay(
+            new Runnable()
+            {
+              @Override
+              public void run()
+              {
+                lookupMgmtLoop();
               }
-              if (allLookupTiers == null) {
-                LOG.info("Not updating lookups because no data exists");
-                return;
+            },
+            2000,
```

Comment:
added a comment to explain, extracting in a constant is not required in this case... it is only used here